src/main/java/main_builders_directors/SalesReporterFeatureBuilder.java

```diff
@@ -0,0 +1,77 @@
+package main_builders_directors;
+
+import interface_adaptors.gateway.GatewayReader;
+import interface_adaptors.sales_report.SalesReporterController;
+import interface_adaptors.sales_report.SalesReporterPresenter;
+import interface_adaptors.sales_report.SalesReporterView;
+import interface_adaptors.show_history.ShowHistoryFileDataAccess;
+import screens.SalesReportScreen;
+import use_cases.gateway_interfaces.GatewayReaderInterface;
+import use_cases.sales_report.SalesReporter;
+import use_cases.sales_report.SalesReporterInputBoundary;
+import use_cases.sales_report.SalesReporterOutputBoundary;
+import use_cases.show_history_use_case.ShowHistoryDsGateway;
+
+import javax.swing.*;
+import java.io.IOException;
+
+public class SalesReporterFeatureBuilder implements FeatureBuilder, GatewayBuilder {
```

Comment:
Great implementation of the builder for the Sales Reporter. This will help clean up main, and allow us to more easily extend new plugins in the future